edx_notifications/base_data.py

```diff
@@ -0,0 +1,287 @@
+"""
+Base objects that data.py uses
+"""
+
+import json
+import copy
+import inspect
+import dateutil.parser
+
+from datetime import datetime
+
+from weakref import WeakKeyDictionary
+
+
+class TypedField(object):
```

Comment:
Approach-wise, if there's only ever a single store per-process, would it be better to force each store to provide the relevant fields, and then just use the store's built in field types? That way you wouldn't need to convert between the store and the base data object. Instead, the store would provide the base data objects themselves.